ros_cross_compile/data_collector.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Classes for time series data collection and writing said data to a file."""
+
+from datetime import datetime
+import json
+from pathlib import Path
+import time
+from typing import NamedTuple, Union
+
+
+class Datum(NamedTuple):
+    """Represents a collected piece of data."""
+
+    name: str
+    value: Union[int, float]
+    unit: str
+    timestamp: str
+
+
+class DataCollector:
+    """Provides an interface to collect time series data."""
+
+    default_file = Path(datetime.now().strftime('%s') + '.json')
+
+    def __init__(self):
+        self._data = []
+
+    def add_datum(self, new_datum: Datum):
+        self._data.append(new_datum._asdict())
+
+    def set_write_path(self, write_path: Path):
```

Comment:
we don't use getters and setters like this in python. why not have this be an argument to the constructor? Or, even better, perhaps it's just an optional argument to `write_file` (reduces statefulness)

Reply:
move writing logic to another class?